src/nn/ops/resample2d.ts

```diff
@@ -49,48 +55,37 @@ export class Resample extends SingleOutputOperation {
     let input: tf.Tensor4D = inputTensors.get(this.input_) as tf.Tensor4D;
     utils.assert(input.rank === 4, 'The input tensor is not 4-D.');
     const sizes: [number, number] = [0, 0];
-    let transposed = false;
     if (this.sizes_ !== undefined) {
-      if (this.sizes_[0] === input.shape[0] &&
-          this.sizes_[1] === input.shape[1]) {
-        sizes[0] = this.sizes_[2];
-        sizes[1] = this.sizes_[3];
-        // assume nchw -> nhwc
-        input = tf.transpose(input, [0, 2, 3, 1]);
-        transposed = true;
-      } else if (
-          this.sizes_[0] === input.shape[0] &&
-          this.sizes_[3] === input.shape[3]) {
-        // assume nhwc
-        sizes[0] = this.sizes_[1];
-        sizes[1] = this.sizes_[2];
-      } else {
-        throw new Error(
-            'tf.image.resize doesn\'t support the sizes parameter.');
-      }
+        // ignore scales
+        sizes[0] = this.sizes_[0];
+        sizes[1] = this.sizes_[1];
     } else if (this.scales_ !== undefined) {
-      if (this.scales_[0] === 1.0 && this.scales_[1] === 1.0) {
-        sizes[0] = Math.floor(input.shape[2] * this.scales_[2]);
-        sizes[1] = Math.floor(input.shape[3] * this.scales_[3]);
-        // assume nchw -> nhwc
-        input = tf.transpose(input, [0, 2, 3, 1]);
-        transposed = true;
-      } else if (this.scales_[0] === 1.0 && this.scales_[3] === 1.0) {
-        // assume nhwc
-        sizes[0] = Math.floor(input.shape[1] * this.scales_[1]);
-        sizes[1] = Math.floor(input.shape[2] * this.scales_[2]);
+      if (this.axes_[0] === 2 && this.axes_[1] === 3) {
+        // nchw
+        sizes[0] = Math.floor(input.shape[2] * this.scales_[0]);
+        sizes[1] = Math.floor(input.shape[3] * this.scales_[1]);
+      } else if (this.axes_[0] === 1 && this.axes_[1] === 2) {
+        // nhwc
+        sizes[0] = Math.floor(input.shape[1] * this.scales_[0]);
+        sizes[1] = Math.floor(input.shape[2] * this.scales_[1]);
+      } else if (this.axes_[0] === 0 && this.axes_[1] === 1) {
+        sizes[0] = Math.floor(input.shape[0] * this.scales_[0]);
+        sizes[1] = Math.floor(input.shape[1] * this.scales_[1]);
       } else {
-        throw new Error(
-            'tf.image.resize doesn\'t support the scales parameter.');
+        throw new Error('axes parameter is invalid.');
       }
     }
+    if (this.axes_[0] === 2 && this.axes_[1] === 3) {
+      // nchw -> nhwc
+      input = tf.transpose(input, [0, 2, 3, 1]);
+    }
```

Comment:
Please also handle `axes_[0] === 0 && axes_[1] === 1` case.

Reply:
@huningxin Is there an actual case? For axes [0, 1], it means HWXX, but I don't know which X is for N, and which X is for C, so it's hard to transpose to target data.  Any suggestion? Thanks.